solana/solitaire/program/src/lib.rs

```diff
@@ -72,10 +73,15 @@ pub enum CreationLamports {
 
 impl CreationLamports {
     /// Amount of lamports to be paid in account creation
-    pub fn amount(self, size: usize) -> u64 {
+    pub fn amount(self, size: usize) -> Result<u64>  {
         match self {
-            CreationLamports::Exempt => Rent::default().minimum_balance(size),
-            CreationLamports::Amount(v) => v,
+            // Rent::get() is used over Rent::default() here in case on-chain Rent has diverged from
+            // the Solana defaults. Unlike `default()`, the `get()` call only works on-chain and in
+            // Solana's test framework, it will fail in unit tests. In those environments one can use
+            // the following:
+            // `CreationLamports::Amount(Rent::default().minimum_balance(size))`
```

Comment:
Can we just add `#[cfg(test)]` and `#[cfg(not(test))]` branches here?  Otherwise it's going to be very difficult for someone to figure out why their tests are failing unless they know to look here.

Reply:
I think letting this bubble up to the user is better than having conditionally compiled behaviour. We have it in some places in the program already and It’s just as easy to trip someone up. While most developers are only targetting Solana itself, allowing users to test against `Rent::default` allows them to continue to write programs that could potentially break later.

If we want to make sure there’s discoverability perhaps we can add some sort of `WormholeError::RentUnavailable` error to WormholeError and bubble that up so that it isn’t surprising behaviour?